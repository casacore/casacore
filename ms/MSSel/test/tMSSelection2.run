#!/bin/sh
# -----------------------------------------------------------------------------
# Usage: tMSSelection.run
# -----------------------------------------------------------------------------
# This script executes the program tMSSelection to test the
# MSSelection module.
#
# $Id$
#-----------------------------------------------------------------------------

  MS=$CASADEMO"mssel_test_small_multifield_spw.ms"

  runCmd(){
      # Uncomment following line to generate an output close to a
      # script to run the tests
      #echo "#Test:$1"; echo $3;
      
      eval "$2 $3"
  }

  getTestMS() {
      tar zxf $testsrcdir/$MS.tgz
  }

  cleanup() {
      \rm -rf $MS
  }

  # Get the test MS (un-tar it from $testsrcdir).
  getTestMS;

  runCmd 1 "$casa_checktool" " ./tMSSelection ms=$MS field=1,2";
  runCmd 2 "$casa_checktool" " ./tMSSelection ms=$MS field=\"2\",2";
  runCmd 3 "$casa_checktool" " ./tMSSelection ms=$MS field=\"2\",1";
  runCmd 4 "$casa_checktool" " ./tMSSelection ms=$MS field=\"2\",1,2000";
  runCmd 5 "$casa_checktool" " ./tMSSelection ms=$MS field=\"2\",1,2000";
  runCmd 6 "$casa_checktool" " ./tMSSelection ms=$MS field=2000";
  runCmd 7 "$casa_checktool" " ./tMSSelection ms=$MS field=20000";

  #
  # Remove the test MS and any other cleanup required.
  #
  cleanup;
