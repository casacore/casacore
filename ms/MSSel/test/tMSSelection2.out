BE: Baseline Expr=
	BE: Ant1         = []
	BE: Ant2         = []
	Baselines = 
FE: Field Expr=1,2
	FE: Field        = [1, 2]
SE: SPW Expr=
	SE: SPW          = []
	SE: Chan         = Axis Lengths: [0, 0] []
	SE: Freq         = Axis Lengths: [0, 0] []
ScE: Scan Expr=
	ScE: tScan         = []
StE: STATE Expr=
	StE: StateObsMode = []
AE: Array Expr=
	AE: Array        = []
TE: Time Expr=
	TE: Time         = Axis Lengths: [3, 0] []
UVE: UVRange Expr=
	UVE: UVRange      = Axis Lengths: [2, 0] []
	UVE: UV in meters = []
OE: Observation Expr=
	OE: ObservationIDList    = []
PE: Poln Expr=
	PE: PolMap       = {}
	PE: CorrMap      = {}

===========================================================
DDIDs(Poln)  = []
DDIDs(SPW)   = []
StateList    = []
Number of selected rows: 3780
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
BE: Baseline Expr=
	BE: Ant1         = []
	BE: Ant2         = []
	Baselines = 
FE: Field Expr=2,2
	FE: Field        = [2]
SE: SPW Expr=
	SE: SPW          = []
	SE: Chan         = Axis Lengths: [0, 0] []
	SE: Freq         = Axis Lengths: [0, 0] []
ScE: Scan Expr=
	ScE: tScan         = []
StE: STATE Expr=
	StE: StateObsMode = []
AE: Array Expr=
	AE: Array        = []
TE: Time Expr=
	TE: Time         = Axis Lengths: [3, 0] []
UVE: UVRange Expr=
	UVE: UVRange      = Axis Lengths: [2, 0] []
	UVE: UV in meters = []
OE: Observation Expr=
	OE: ObservationIDList    = []
PE: Poln Expr=
	PE: PolMap       = {}
	PE: CorrMap      = {}

===========================================================
DDIDs(Poln)  = []
DDIDs(SPW)   = []
StateList    = []
Number of selected rows: 1440
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
BE: Baseline Expr=
	BE: Ant1         = []
	BE: Ant2         = []
	Baselines = 
FE: Field Expr=2,1
	FE: Field        = [1, 2]
SE: SPW Expr=
	SE: SPW          = []
	SE: Chan         = Axis Lengths: [0, 0] []
	SE: Freq         = Axis Lengths: [0, 0] []
ScE: Scan Expr=
	ScE: tScan         = []
StE: STATE Expr=
	StE: StateObsMode = []
AE: Array Expr=
	AE: Array        = []
TE: Time Expr=
	TE: Time         = Axis Lengths: [3, 0] []
UVE: UVRange Expr=
	UVE: UVRange      = Axis Lengths: [2, 0] []
	UVE: UV in meters = []
OE: Observation Expr=
	OE: ObservationIDList    = []
PE: Poln Expr=
	PE: PolMap       = {}
	PE: CorrMap      = {}

===========================================================
DDIDs(Poln)  = []
DDIDs(SPW)   = []
StateList    = []
Number of selected rows: 3780
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
BE: Baseline Expr=
	BE: Ant1         = []
	BE: Ant2         = []
	Baselines = 
FE: Field Expr=2,1,2000
	FE: Field        = [1, 2, 4]
SE: SPW Expr=
	SE: SPW          = []
	SE: Chan         = Axis Lengths: [0, 0] []
	SE: Freq         = Axis Lengths: [0, 0] []
ScE: Scan Expr=
	ScE: tScan         = []
StE: STATE Expr=
	StE: StateObsMode = []
AE: Array Expr=
	AE: Array        = []
TE: Time Expr=
	TE: Time         = Axis Lengths: [3, 0] []
UVE: UVRange Expr=
	UVE: UVRange      = Axis Lengths: [2, 0] []
	UVE: UV in meters = []
OE: Observation Expr=
	OE: ObservationIDList    = []
PE: Poln Expr=
	PE: PolMap       = {}
	PE: CorrMap      = {}

===========================================================
DDIDs(Poln)  = []
DDIDs(SPW)   = []
StateList    = []
Number of selected rows: 5220
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
BE: Baseline Expr=
	BE: Ant1         = []
	BE: Ant2         = []
	Baselines = 
FE: Field Expr=2,1,2000
	FE: Field        = [1, 2, 4]
SE: SPW Expr=
	SE: SPW          = []
	SE: Chan         = Axis Lengths: [0, 0] []
	SE: Freq         = Axis Lengths: [0, 0] []
ScE: Scan Expr=
	ScE: tScan         = []
StE: STATE Expr=
	StE: StateObsMode = []
AE: Array Expr=
	AE: Array        = []
TE: Time Expr=
	TE: Time         = Axis Lengths: [3, 0] []
UVE: UVRange Expr=
	UVE: UVRange      = Axis Lengths: [2, 0] []
	UVE: UV in meters = []
OE: Observation Expr=
	OE: ObservationIDList    = []
PE: Poln Expr=
	PE: PolMap       = {}
	PE: CorrMap      = {}

===========================================================
DDIDs(Poln)  = []
DDIDs(SPW)   = []
StateList    = []
Number of selected rows: 5220
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
BE: Baseline Expr=
	BE: Ant1         = []
	BE: Ant2         = []
	Baselines = 
FE: Field Expr=2000
	FE: Field        = [4]
SE: SPW Expr=
	SE: SPW          = []
	SE: Chan         = Axis Lengths: [0, 0] []
	SE: Freq         = Axis Lengths: [0, 0] []
ScE: Scan Expr=
	ScE: tScan         = []
StE: STATE Expr=
	StE: StateObsMode = []
AE: Array Expr=
	AE: Array        = []
TE: Time Expr=
	TE: Time         = Axis Lengths: [3, 0] []
UVE: UVRange Expr=
	UVE: UVRange      = Axis Lengths: [2, 0] []
	UVE: UV in meters = []
OE: Observation Expr=
	OE: ObservationIDList    = []
PE: Poln Expr=
	PE: PolMap       = {}
	PE: CorrMap      = {}

===========================================================
DDIDs(Poln)  = []
DDIDs(SPW)   = []
StateList    = []
Number of selected rows: 1440
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
###AipsError: Field Expression: Partial or no match for Field ID list [20000]
(near char. 5 in string "20000")
